### PR TITLE
feat(datagrid): Support multi column selection

### PR DESCRIPTION
**Pull request by gjou-tlnd**
**What is the problem this PR is trying to solve?**

- Support multi column selection.
- The main components were migrated to typescript
- Better separation of concern between data mapping & grid rendering
  - To render a dataset sample, use `DatasetSerializer.getColumnDefs`
  - To render data from another source, build your own `columnDefs`

**What is the chosen solution to this problem?**

**Please check if the PR fulfills these requirements**

- [ ] The PR have used `yarn changeset` to a request a release from the CI if wanted.
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/datagrid/README.md [hunk 1]
```diff
@@ -96,41 +68,6 @@ It is a list of issues encountered with ag-grid that DataGrid resolved.
...
-## Ag-Grid Performance
-
-[ag-grid tips](https://www.ag-grid.com/javascript-grid-performance/)
-
-Although that Ag-Grid recommands to avoid to use custom cell renderer to renderer the cell, we used because a cell can't have only text but many elements on this: quality indicator, comments, invisible chars, ...
-
-This choice creates a issue when we quickly scroll, by the mouse, the scrollbar. With the scroll whell, the performance is acceptable.
-
-For IE11, we use Graceful Degradation.
+Ag-grid set the current cell selected by a click. If we navigate with the keyboard, ag-grid sent a new event onCellFocused but the cell isn't selected. AgGrid allows to change this behavior by adding a method _navigateToNextCell_ on theses options. Each times that we move to a next cell, this cell is selected. The current selected rows is hightlight by the class _.ag-row-focus_. This class can be overrided by the class _.ag-row-selected_ is many row is selected.
```
Comment (Benrajalu) [inline packages/datagrid/README.md:73]:
```suggestion
Ag-grid sets the current cell selected by a click. If we navigate with the keyboard, ag-grid sends a new event onCellFocused but the cell isn't selected. AgGrid allows to change this behavior by adding a method _navigateToNextCell_ on these options. Each time that we move to a next cell, this cell is selected. The current selected rows is highlighted by the class _.ag-row-focus_. This class can be overridden by the class _.ag-row-selected_ if many row are selected.
```

File: packages/datagrid/src/components/DataGrid/DataGrid.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+			setSelectedColumns(controlledColumnIds);
+		}
+	}, [controlledColumnIds]);
+
+	// Force style update when selection changed
+	useEffect(() => {
+		const previousSelection = context.current.selectedColumns;
+		context.current.selectedColumns = selectedColumns;
+
+		// BEGIN QUICKFIX: Would be better to use gridRef.current?.api.refreshHeader() but it has a weird flicker
```
Comment (Benrajalu) [inline packages/datagrid/src/components/DataGrid/DataGrid.tsx:99]: What if you set the display of the grid (through opacity, not display: none) to a timer? You keep a skeleton running above for the duration of that timer. Fade out the skeleton / fade in the grid with refreshHeader?

Comment (Benrajalu) [reply packages/datagrid/src/components/DataGrid/DataGrid.tsx:99]: Discussed IRL, that's not it.

File: packages/datagrid/src/serializers/datasetSerializer.test.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import sample from '../sample.json';
+import { getColumnDefs } from './datasetSerializer';
+
+describe('#getColumnDefs', () => {
```
Comment (sgangnant) [inline packages/datagrid/src/serializers/datasetSerializer.test.ts:4]: Why the hashtag ?

```suggestion
describe('getColumnDefs', () => {
```

**Review by sgangnant — APPROVED** [review]

File: packages/datagrid/src/serializers/datasetSerializer.test.ts (hunk 3 above)
Comment (gjou-tlnd) [reply packages/datagrid/src/serializers/datasetSerializer.test.ts:4]: dunno, legacy code

File: packages/datagrid/src/components/DataGrid/DataGrid.tsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
+import React, { useEffect, useMemo, useRef } from 'react';
+
+import { CellFocusedEvent, GridOptions, ColDef } from 'ag-grid-community';
+import { AgGridReact } from 'ag-grid-react';
```
Comment (jmfrancois) [inline packages/datagrid/src/components/DataGrid/DataGrid.tsx:4]: lazy load please :pray:

**Review by jmfrancois — CHANGES_REQUESTED** [review]

File: packages/datagrid/src/components/DataGrid/DataGrid.tsx (hunk 4 above)
Comment (gjou-tlnd) [reply packages/datagrid/src/components/DataGrid/DataGrid.tsx:4]: I need this import, it's used as a type

**Review by jmfrancois — APPROVED** [review]

**Pull request merged**